src/pyflex/window_selector.py

```diff
@@ -316,25 +339,28 @@ def select_windows(self):
 
         self.calculate_preliminiaries()
 
+        self.determine_signal_and_noise_indices()
+        if self.config.check_global_data_quality:
+            if not self.check_data_quality():
+                return []
+
```

Comment:
I have written a method in config.py that convert negative index into positive index and then do the checking.